### PR TITLE
Feature/179 scaffold

### PR DESCRIPTION
**Pull request by algattik**
Creates scaffold Registry service web service as an EDC runtime with simple health endpoint. 

CI pipeline builds the registration service and runs it in docker. A system test runs an HTTP request as a client to validate the build.

- The GitHub action workflows are taken from MVD
- The OpenAPI generation task is taken from EDC
- Then REST client generation task is taken from spike https://github.com/agera-edc/DataSpaceConnector/pull/268
- Gradle wrapper, Dockerfile and docker-compose structure are taken from MVD

Closes https://github.com/agera-edc/MinimumViableDataspace/issues/179

On naming, we use `org.eclipse.dataspaceconnector.registration` since the repo will live in the https://github.com/eclipse-dataspaceconnector organization.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# RegistrationService
+Temporary repository to get started with the registration service for the MVD
```
Comment (marcgs) [inline README.md:2]:
```suggestion
Temporary repository to get started with the registration service for the MVD
```

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            outputFormat = io.swagger.v3.plugins.gradle.tasks.ResolveTask.Format.YAML
+            prettyPrint = true
+            classpath = java.sourceSets["main"].runtimeClasspath
+            buildClasspath = classpath
+            resourcePackages = setOf("org.eclipse.dataspaceconnector")
+            outputDir = file("${rootProject.projectDir.path}/resources/openapi/yaml")
+        }
+        configurations {
+            all {
+                exclude(group = "com.fasterxml.jackson.jaxrs", module = "jackson-jaxrs-json-provider")
```
Comment (marcgs) [inline build.gradle.kts:37]: Do you know why is this required?

File: docs/developer/openapi.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+task.
+
+The file is at `resources/openapi/yaml/registration-service.yaml`.
+
+## Generate OpenAPI `yaml` file
+
+To re-generate the YAML file, invoke 
+```shell
+./gradlew clean resolve
+```
```
Comment (cpeeyush) [inline docs/developer/openapi.md]: nit: we can also add that how to generate rest client from this yaml file.

File: extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/health/HealthApiController.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.Response;
+
+@Tag(name = "Health")
+@Produces({"application/json"})
+@Consumes({"application/json"})
+@Path("/health")
+public class HealthApiController {
```
Comment (marcgs) [inline extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/health/HealthApiController.java:17]: Did you consider using EDC's `ObservabilityApiController`?

**Review by marcgs — APPROVED** [review]

File: system-tests/src/test/java/org/eclipse/dataspaceconnector/registration/client/IntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+@EnabledIfEnvironmentVariable(named = "INTEGRATION_TEST", matches = "true")
```
Comment (cpeeyush) [inline system-tests/src/test/java/org/eclipse/dataspaceconnector/registration/client/IntegrationTest.java:26]: - we can add a follow up story to add doc about details on how to run integration test. 
- And also, a system test README explaining that project needs to be build first to generate executable jar `./build/libs/app.jar ` and then only docker-compose will work locally.

Both of these items can be done in a follow-up story.

File: build.gradle.kts (hunk 2 above)
Comment (algattik) [reply build.gradle.kts:37]: no, but that's how it is in EDC and removing it gives a missing dependency error

File: docs/developer/openapi.md (hunk 3 above)
Comment (algattik) [reply docs/developer/openapi.md]: done

File: system-tests/src/test/java/org/eclipse/dataspaceconnector/registration/client/IntegrationTest.java (hunk 5 above)
Comment (algattik) [reply system-tests/src/test/java/org/eclipse/dataspaceconnector/registration/client/IntegrationTest.java:26]: Added system test README.md

File: extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/health/HealthApiController.java (hunk 4 above)
Comment (algattik) [reply extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/health/HealthApiController.java:17]: Good point, I did want to add a custom controller to show how it's done, but this can be a follow-up story after we have more controllers.

File: system-tests/src/test/java/org/eclipse/dataspaceconnector/registration/client/HealthApiClientTest.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import org.eclipse.dataspaceconnector.registration.client.api.HealthApi;
+import org.junit.jupiter.api.Test;
+
+import java.time.Duration;
+
+import static org.awaitility.Awaitility.await;
+
+@IntegrationTest
+public class HealthApiClientTest {
+    static final String API_URL = "http://localhost:8181/api";
```
Comment (cpeeyush) [inline system-tests/src/test/java/org/eclipse/dataspaceconnector/registration/client/HealthApiClientTest.java:26]: very nit: maybe we can read it from environment variables and keep this one as default. or we can do it later as well when we extend the test to verify health endpoint of a deployed one.

File: extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/health/HealthApiController.java (hunk 4 above)
Comment (algattik) [reply extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/health/HealthApiController.java:17]: Created https://github.com/agera-edc/MinimumViableDataspace/issues/203

File: extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/RegistrationServiceApiExtension.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package org.eclipse.dataspaceconnector.registration.api;
+
+import org.eclipse.dataspaceconnector.registration.api.health.HealthApiController;
+import org.eclipse.dataspaceconnector.spi.WebService;
+import org.eclipse.dataspaceconnector.spi.system.Inject;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+
+public class RegistrationServiceApiExtension implements ServiceExtension {
```
Comment (Izzzu) [inline extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/RegistrationServiceApiExtension.java:12]: Should we add javadoc?

**Review by cpeeyush — APPROVED** [review]
Nice work!

File: system-tests/src/test/java/org/eclipse/dataspaceconnector/registration/client/HealthApiClientTest.java (hunk 6 above)
Comment (algattik) [reply system-tests/src/test/java/org/eclipse/dataspaceconnector/registration/client/HealthApiClientTest.java:26]: Yes I first had a version with env var, but then I removed it along the principle to only implement things once we have a clear need for them :)

File: extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/RegistrationServiceApiExtension.java (hunk 7 above)
Comment (algattik) [reply extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/RegistrationServiceApiExtension.java:12]: added
